### DISCUSSION TITLE
Bank number and program number

### DISCUSSION THREAD

**Discussion by js-midi** [shanteacontrols/OpenDeck#155] [Ideas]
hello.
Always grateful for the project.
I would like to suggest a button function.
I wonder what it would be like to be able to set both a program number and a bank number on a button at the same time.
I made a nice box for changing the program of the sound module.
It is very convenient when playing live.
A little disappointing part is that the bank cannot be loaded with the program.

Thanks for reading.
![20220602_220934](https://user-images.githubusercontent.com/80445117/171638277-44fd581d-8fa5-463a-8650-1e940aa90489.jpg)

**Comment by paradajz** [comment]
I'm sorry but I don't really understand what you're asking here.

**Reply by js-midi** [reply]
Sorry for my lack of English
Please look at the picture and understand
![opendecknank](https://user-images.githubusercontent.com/80445117/171739586-88917ac7-d58d-404c-96a1-6d87a3bebfa8.jpg)

**Comment by paradajz** [comment]
What is "bank" in this context? An OpenDeck preset? Or something else?

**Comment by js-midi** [comment]
This is the bankchange command in the midi message.
What is in the picture is my graphic drawing

**Comment by paradajz** [comment]
Ah, so this then: http://midi.teragonaudio.com/tutr/bank.htm

I wasn't aware this even existed. So, you want to be able to map a button to send bank change message?

**Reply by js-midi** [reply]
thanks for the quick response
I want to map program change and bank change to a button, but I don't have that ability.
Thank you again

**Comment by paradajz** [comment]
According to the page I've linked above, bank change is composed of CC message and program change message, which should get interpreted as bank change. So there's two messages sent at once.

![image](https://user-images.githubusercontent.com/2544094/171798495-21a6ae41-057f-4414-93b4-49b5cbcf7b52.png)

Please confirm that you want this.

**Reply by js-midi** [reply]
if you could make this possible
Please, this is the function I want

**Reply by dolsoidduk** [reply]
> According to the page I've linked above, bank change is composed of CC message and program change message, which should get interpreted as bank change. So there's two messages sent at once.
> 
> ![image](https://user-images.githubusercontent.com/2544094/171798495-21a6ae41-057f-4414-93b4-49b5cbcf7b52.png)
> 
> Please confirm that you want this.

want this

**Comment by js-midi** [comment]
yes i know that too
I know you send bank change commands with msb and lsb
ex) Bank change cc=0 I know that the desired bank 0~127 is like this

**Reply by paradajz** [reply]
So is that what you want then? Or you want bank change *and* another unrelated program change message? The latter isn't possible but the former can be added.

**Comment by js-midi** [comment]
I want to map bank change and program change together
I don't know if it's possible, but...
Thanks for the kind reply and help

**Reply by paradajz** [reply]
That's not possible. It's one message per component, so just program change or just bank change.

**Comment by js-midi** [comment]
Currently, you can change the bank with the cc command. It is possible with midi id and Value values.
But I wanted to know if it is possible to map two commands at the same time.
If it's impossible, I'll give up ^^
Thank you.

**Comment by js-midi** [comment]
I think I misunderstood
It seems that the bank change includes a program change.
In this case, I think I will be
thank you

**Comment by js-midi** [comment]
I think I will be very happy if the bank change function is implemented.
thank you

**Comment by m0rgano** [comment]
I think the ability to only be able to send one 'message' on a button is a serious shortcoming.

As the OP describes, a Program change often requires a preceding Bank change, especially on devices with more than 128 program slots.

Really, it would be good if an 'unlimited' number of messages could be triggered by a button, and on different MIDI channels too, as there may be more than one receiving MIDI device connected in a chain.

**Reply by paradajz** [reply]
> I think the ability to only be able to send one 'message' on a button is a serious shortcoming.

Connect one button to multiple inputs on board in parallel and you've solved the problem.
